File: python_solution/problem8.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Mon Aug 19 19:30:51 2024

@author: sharminalina
"""
class Problem8:
    def findIntegerfromString(self, s: str)-> str:
        i = 0
        n = len(s)
        sign = 1
        result = 0

        while i< n and s[i] == ' ':
            i +=1
            if (i < n and (s[i]== '-' or s[i]== '+')):
                sign = -1 if s[i]== '-' else 1
                i += 1
            if(i < n and (s[i] == '-' or  s[i] == '+')):
                result = 0
    
        while i < n and s[i].isdigit():
            digit = int(s[i])
            if result > (2**31 - 1 - digit) // 10:
                result=2**31 - 1 if sign == 1 else -2**31            

            result = result * 10 + digit
            i += 1

        return result*sign

def main():
    # Read input from the console
   
    x= input("Enter String to convert Into integer : ")

    # Create an instance of Solution and find the Integer from the String
    solution_prob8 = Problem8()
    result = solution_prob8.findIntegerfromString(x)

    # Print the result
    print(f"Integer Number of String {x} is : {result}")

if __name__ == "__main__":
    main()

File: C#Solutions/problem8.cs
using System;

namespace AllProblems
{
    public static class Problem8
    {
        public static void Solve()
        {
            Console.WriteLine("Solution to Problem 8");
            
            Console.WriteLine("Enter a string to convert into intiger:");
            string s = Console.ReadLine();
            int result = convertStringtoIntiger(s);
            
            Console.WriteLine("Number converted from string: " + result);
            
        }

        

        public static int convertStringtoIntiger(string s)
        {
            int i = 0;
            int n = s.Length;
            int result = 0;
            int sign = 1;
            while(i < n && s[i] == ' '){
                i++;
            }
            if(i < n  && (s[i]== '-'||s[i]== '+')){
                sign = s[i] == '-'? -1: 1;
                i++;
            }
            if(i < n  && (s[i]== '-'||s[i]== '+')){
                return 0;
            }
            while(i< n && char.IsDigit(s[i]) ){
                int digit = s[i] -'0';
                if (result > (int.MaxValue - digit) / 10)
                {
                    return sign == 1 ? int.MaxValue : int.MinValue;
                }
                result = result*10 + digit;
                i++;
            }
        
            return result*sign;
            
        }

        
    }
}
